x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/InternalUsers.java

```diff
@@ -7,94 +7,144 @@
+                    .build() },
+            new String[] {},
+            MetadataUtils.DEFAULT_RESERVED_METADATA
+        )
+    );
+
+    /**
+     * XPack internal user that manages xpack. Has all cluster/indices permissions for x-pack to operate excluding security permissions.
+     */
+    public static final InternalUser XPACK_USER = new InternalUser(
+        UsernamesField.XPACK_NAME,
+        new RoleDescriptor(
+            UsernamesField.XPACK_ROLE,
+            new String[] { "all" },
+            new RoleDescriptor.IndicesPrivileges[] {
+                RoleDescriptor.IndicesPrivileges.builder()
+                    .indices("/@&~(\\.security.*)&~(\\.async-search.*)/")
+                    .privileges("all")
+                    .allowRestrictedIndices(true)
+                    .build() },
+            new String[] { "*" },
+            MetadataUtils.DEFAULT_RESERVED_METADATA
+        )
+    );
+
+    /**
+     * internal user that manages xpack security. Has all cluster/indices permissions.
+     */
+    public static final InternalUser XPACK_SECURITY_USER = new InternalUser(
+        UsernamesField.XPACK_SECURITY_NAME,
+        new RoleDescriptor(
+            UsernamesField.XPACK_SECURITY_ROLE,
+            new String[] { "all" },
+            new RoleDescriptor.IndicesPrivileges[] {
+                RoleDescriptor.IndicesPrivileges.builder().indices("*").privileges("all").allowRestrictedIndices(true).build() },
+            null,
+            null,
+            new String[] { "*" },
+            MetadataUtils.DEFAULT_RESERVED_METADATA,
+            Map.of()
+        )
+    );
+
+    private static final Map<String, InternalUser> INTERNAL_USERS = new HashMap<>();
 
-    private static final Map<String, UserInstance> INTERNAL_USERS = new HashMap<>();
     static {
-        defineUser(
-            SystemUser.NAME,
-            SystemUser.INSTANCE,
-            null /* SystemUser relies on a simple action predicate rather than a role descriptor */,
-            SystemUser::is
-        );
-        defineUser(XPackUser.NAME, XPackUser.INSTANCE, XPackUser.ROLE_DESCRIPTOR, XPackUser::is);
-        defineUser(XPackSecurityUser.NAME, XPackSecurityUser.INSTANCE, XPackSecurityUser.ROLE_DESCRIPTOR, XPackSecurityUser::is);
-        defineUser(SecurityProfileUser.NAME, SecurityProfileUser.INSTANCE, SecurityProfileUser.ROLE_DESCRIPTOR, SecurityProfileUser::is);
-        defineUser(AsyncSearchUser.NAME, AsyncSearchUser.INSTANCE, AsyncSearchUser.ROLE_DESCRIPTOR, AsyncSearchUser::is);
-        defineUser(
-            CrossClusterAccessUser.NAME,
-            CrossClusterAccessUser.INSTANCE,
-            null /* CrossClusterAccessUser has a role descriptor, but it should never be resolved by this class */,
-            CrossClusterAccessUser::is
-        );
-        defineUser(StorageInternalUser.NAME, StorageInternalUser.INSTANCE, StorageInternalUser.ROLE_DESCRIPTOR, StorageInternalUser::is);
+        defineUser(SystemUser.INSTANCE);
```

Comment:
I don't have a strong view either way. I can see an argument for all internal users being in 1 place, but it means the `SystemUser` constructor needs to become package protected instead of private.

Reply:
A middle ground is probably just create a copy for `SystemUser.INSTANCE` in `InternalUsers`, i.e. `InteralUsers.SYSTEM_USER = SystemUser.INSTANCE`. What do you think?